lua/kulala/ui/init.lua

```diff
@@ -118,6 +119,28 @@ local function pretty_ms(ms)
   return string.format("%.2fms", ms)
 end
 
+---Prints the parsed Request table into current buffer - uses nvim_put
+local function print_http_spec(spec)
+  local lines = {}
+  local idx = 1
+
+  lines[idx] = spec.method .. " " .. spec.url
+  if spec.http_version ~= "" then
+    lines[idx] = lines[idx] .. " " .. spec.http_version
+  end
+  for header, value in pairs(spec.headers) do
+    idx = idx + 1
+    lines[idx] = header .. ": " .. value
+  end
+  if spec.body ~= "" then
+    idx = idx + 1
+    lines[idx] = ""
+    -- FIXME: broken for multi-line body
+    lines[idx+1] = spec.body
+  end
+  vim.api.nvim_put(lines, 'l', false, false)
```

Comment:
Dunno if this is a good approach, however is easier to use than nvim_buf_set_lines, which overwrite the content of the file.